t/APISIX.pm

```diff
@@ -506,8 +503,11 @@ _EOC_
     end
 
     require "resty.core"
-
-    $extra_init_by_lua_start
+_EOC_
+    my $calling_sub = (caller(1))[1];
+    if (index($calling_sub, "tars") != -1) {
```

Comment:
It looks like we didn't actually remove the `extra_init_by_lua_start` mount point. It's just determining if it's a tars module in the APISIX.pm file and mounting it.

If there are other test cases that we want to mount in this location in the future, we need to add it in APISIX.pm?

Reply:
IIUC the code, tars module `extra_init_by_lua` code need to load before APISIX init ,other modules `extra_init_by_lua` code  need to load after APISIX init.If we must remove `extra_init_by_lua_start`, we need to determine the execution order of `extra_init_by_lua` in a way similar to judging the module name.This way of displaying declarations in APISIX.pm may be more explicit.

Reply:
Actually, I don't think this is better than the previously rejected solution in https://github.com/apache/apisix/pull/8254#discussion_r1017453415...